File: packages/shared/domain/schema.ts
import { z } from 'zod';

// P18 (v1.1) D2: the wire shape for a connection's pasted DDL document — the raw text only. The
// parsed table/column model the SQL language service actually completes/diagnoses/hovers against
// (views/console/ddl.ts's DdlSchema) is derived client-side from this text and never crosses the
// wire itself.
export const connectionDdlSchema = /*#__PURE__*/ z.object({
  connectionId: z.string(),
  ddl: z.string(),
  /** ISO timestamp of the last Save; '' for a connection with no saved document (D2: absent
   *  until the user writes one, not an error). */
  updatedAt: z.string(),
});
export type ConnectionDdl = z.infer<typeof connectionDdlSchema>;

File: apps/kira-studio/internal/storage/model/schema.go
package model

// ConnectionDDL mirrors packages/shared/domain/schema.ts's connectionDdlSchema — the DDL text a
// user pastes for one connection so the SQL language service can complete/diagnose/hover against
// it (P18 v1.1 D2). ConnectionID is carried on the wire type (not just used as a lookup key) so a
// SchemaService.Set broadcast can be applied by connectionId with no extra field.
type ConnectionDDL struct {
	ConnectionID string `json:"connectionId"`
	DDL          string `json:"ddl"`
	UpdatedAt    string `json:"updatedAt"`
}
